src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
> this is not important in D, because there are only interfaces without data members. Any instance of the vtbl that contains the correct function pointers will do.

You're right. That's the advantage of single inheritance model.

Anyway such the change is in scope of optimization, and would need more thoughts.